site/public/js/rainbow-customization.js

```diff
@@ -267,78 +267,144 @@ function getGradeableBuckets() {
     return gradeables;
 }
 
-function getPlagiarism() {
-    const plagiarismData = [];
+/**
+ * Gets data to add to the JSON for the plagiarism and manual grade tables
+ * @param {int} table
+ *      0 -> plagiarism table
+ *      1 -> manual grading table
+ */
+function getTableData(table) {
+    if (table !== 0 && table !== 1) {
+        return;
+    }
+
+    const data = [];
 
-    const tableBody = document.getElementById('table-body');
+    const tableMap = {
+        0: 'plagiarism-table-body',
+        1: 'manual-grading-table-body',
+    };
+    const tableBody = document.getElementById(tableMap[table]);
     const rows = tableBody.getElementsByTagName('tr');
 
     for (let i = 0; i < rows.length; i++) {
         const row = rows[i];
-        const user = row.cells[0].textContent;
-        const gradeable = row.cells[1].textContent;
-        const penalty = parseFloat(row.cells[2].textContent);
-
-        plagiarismData.push({
-            user: user,
-            gradeable: gradeable,
-            penalty: penalty,
-        });
+        const firstInput = row.cells[0].textContent;
+        const secondInput = row.cells[1].textContent;
+        const thirdInput = row.cells[2].textContent;
+
+        if (table === 0) {
+            data.push({
+                user: firstInput,
+                gradeable: secondInput,
+                penalty: parseFloat(thirdInput),
+            });
+        }
+        else if (table === 1) {
+            data.push({
+                user: firstInput,
+                grade: secondInput,
+                note: thirdInput,
+            });
+        }
```

Comment:
These can be combined, with a ternary used to determine the value of the `note` field.

Reply:
I was thinking about this too, but we will be adding more tables to it later on, so it should be ok.